src/Shared/FileUtilities.cs

```diff
@@ -836,5 +837,36 @@ internal static string CombinePaths(string root, params string[] paths)
 
             return paths.Aggregate(root, Path.Combine);
         }
+
+        internal static string TrimTrailingSlashes(this string s)
+        {
+            return s.TrimEnd('/', '\\');
+        }
+
+        /// <summary>
+        /// Replace all backward slashes to forward slashes
+        /// </summary>
+        internal static string ToSlash(this string s)
+        {
+            return s.Replace('\\', '/');
+        }
+
+        internal static string NormalizeForPathComparison(this string s) => s.ToSlash().TrimTrailingSlashes();
+
+        internal static bool PathsEqual(string path1, string path2)
```

Comment:
In an ideal world this wouldn't be so allocatey. We should keep an eye out--if we see increased GC pressure, this could be a culprit (depends how often it's called).